app/backend/src/couchers/servicers/api.py

```diff
@@ -525,12 +526,9 @@ def ListMutualFriends(self, request, context):
                 .filter(FriendRelationship.status == FriendStatus.accepted)
             )
 
-            mutual_friends = (
-                session.query(User)
-                .filter_users(context)
-                .filter(User.id.in_(q1.union(q2).intersect(q3.union(q4)).subquery()))
-                .all()
-            )
+            mutual = select(intersect(union(q1, q2), union(q3, q4)).subquery())
```

Comment:
Both extracting this onto it's own line as well as using intersection and union in this manner are not strictly necessary, but I find it clearer and prettier